gptme/config.py

```diff
@@ -156,6 +182,47 @@ def get_workspace_prompt(workspace: str) -> str:
     return ""
 
 
+class Provider(str, Enum):
+    OPENAI = "openai"
+    AZURE_OPENAI = "azure"
+    ANTHROPIC = "anthropic"
+    OPENROUTER = "openrouter"
+    LOCAL = "local"
+
+    def __repr__(self) -> str:
+        return self.value
+
+
+class LLMAPIConfig(BaseModel):
+    endpoint: HttpUrl | None = Field(default=None)
+    token: str
+    provider: Provider
+    model: str | None
+
+    _envvar_api_key: str = "API_KEY"
+    _envvar_provider: str = "API_PROVIDER"
+    _envvar_model: str = "API_MODEL"
+
+    @property
+    def _envvar_endpoint(self) -> str:
+        if self.provider == Provider.OPENAI or self.provider == Provider.AZURE_OPENAI:
+            return "API_ENDPOINT"
+        return ""
+
+    def save_to_config(self):
+        set_config_value(f"env.{self._envvar_api_key}", self.token)
+        set_config_value(f"env.{self._envvar_provider}", self.provider.value)
+        if not self._envvar_endpoint:
+            logger.warning(
+                f"Provider {self.provider.value} has no custom endpoint, skipping saving to config"
+            )
+            return
+        if self.model:
+            set_config_value(f"env.{self._envvar_model}", self.model)
+        if self.endpoint:
+            set_config_value(f"env.{self._envvar_endpoint}", str(self.endpoint))
```

Comment:
Not sure if I understand the motivation behind this, but I am not keen on diverging from the standard environment variable names.

Reply:
I have been using some tools for wrapping the OpenAI SDK, which use their own config files to specify the API key and endpoint. For example, you can see how it's done in [TheR1D/shell_gpt](https://github.com/TheR1D/shell_gpt). I believe it's a good practice to make it easy for users to use the SDK without having to dig into the source code to figure out what "key" should be put into the config file.

Reply:
>I believe it's a good practice to make it easy for users to use the SDK without having to dig into the source code to figure out what "key" should be put into the config file.

Users are prompted for a key on first start, and the provider for the key is auto-detected. I don't think this is easier, it's a lot more code.

Reply:
Some proxy sites do not provide the key following the pattern set by OpenAI. For example, they may not provide the API key with the `sk-` prefix. In this case, the "detection" mechanism will get stuck in an infinite loop, continuously prompting for an invalid-API-key error.